lib/ansible/modules/network/aos/aos_device.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
+#
+# (c) 2017 Apstra Inc, <community@apstra.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: aos_device
+author: Damien Garros (@dgarros)
+version_added: "2.3"
+short_description: Manage Devices on AOS Server
+description:
+  - Apstra AOS Device module let you manage your devices in AOS easily. You can
+    approve devices and define in which state the device should be. Currently
+    only the state I(normal) is supported but the goal is to extend this module
+    with additional state. This module is idempotent and support the I(check) mode.
+    It's using the AOS REST API.
+requirements:
+  - "aos-pyez >= 0.6.0"
+options:
+  session:
+    description:
+      - An existing AOS session as obtained by aos_login module
+    required: true
+  name:
+    description:
+      - The device serial-number; i.e. uniquely identifies the device in the
+        AOS system. Only one of I(name) or I(id) can be set.
+  id:
+    description:
+      - The AOS internal id for a device; i.e. uniquely identifies the device in the
+        AOS system. Only one of I(name) or I(id) can be set.
+  state:
+    description:
+      - Define in which state the device should be. Currently only I(normal)
+        is supported but the goal is to add I(maint) and I(decomm).
+    default: normal
+    choices: ['normal']
+  approve:
+    description:
+      - The approve argument instruct the module to convert a device in quarantine
+        mode into approved mode.
+  location:
+    description:
+      - When approving a device using the I(approve) argument, it's possible
```

Comment:
In `argspec` you could use `required_if` to enforce this. Assuming you want to throw an error if `location` is set and `approve` isn't